### PR TITLE
Feature/unit tests/tags parser

### PR DESCRIPTION
**Pull request by kkania-splunk**
ut for tags_parser.py + eventtype_parser ut refactor

### CODE REVIEW

File: tests/unit/test_tags_parser.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    "source=%2Fopt%2Fsplunk%2Fvar%2Flog%2Fsplunk%2Fsplunkd.log": {
+        "tags_positive_event": "enabled",
+        "tags_disabled_event": "disabled",
+    },
+}
+
+
+def test_tags_can_be_parsed_and_extracted(parser_instance):
+    assert hasattr(
+        parser_instance.tags, "sects"
```
Comment (uoboda-splunk) [inline tests/unit/test_tags_parser.py:22]: It looks more like precondition to test. It is possible that `parser_instance.tags == []` for example

File: tests/unit/test_tags_parser.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def test_get_tags_calls_app_get_config(parser_instance):
+    for _ in parser_instance.get_tags():
+        pass
+    parser_instance.app.get_config.assert_called_once()
+
+
+def test_no_tags_config_file(parser_instance):
+    parser_instance.app.get_config.side_effect = OSError
+    output = [tag for tag in parser_instance.get_tags() if tag]
+    assert output == [], "tags created when no config file exists"
```
Comment (uoboda-splunk) [inline tests/unit/test_tags_parser.py:54]:
```suggestion
    assert parser_instance.tags is None, "tags created when no config file exists"
```
Should be enough

File: tests/unit/test_tags_parser.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        stanza = stanza.replace("=", '="')
+        stanza = unquote('{}"'.format(stanza))
+        for item, value in section.options.items():
+            expected_outputs.append(
+                {
+                    "stanza": stanza,
+                    "tag": value.name,
+                    "enabled": True if value.value == "enabled" else False,
+                }
+            )
```
Comment (uoboda-splunk) [inline tests/unit/test_tags_parser.py:38]: It mirrors logic from source code. I think it is better to use raw data samples in assertions. We should avoid code duplications in tests

File: tests/unit/test_tags_parser.py (hunk 2 above)
Comment (kkania-splunk) [reply tests/unit/test_tags_parser.py:65]: I prefer to leave it as it is, because calling `get_tags` instead of just `tags` covers 1 more `if` statement

**Review by uoboda-splunk — APPROVED** [review]

**Pull request merged**